lib/SPIRV/SPIRVRegularizeLLVM.cpp

```diff
@@ -605,6 +612,69 @@ void SPIRVRegularizeLLVMBase::lowerFuncPtr(Module *M) {
     lowerFuncPtr(I.first, I.second);
 }
 
+void SPIRVRegularizeLLVMBase::addKernelEntryPoint(Module *M) {
+  std::vector<Function *> Work;
+
+  // Get a list of all functions that have SPIR kernel calling conv
+  for (auto &F : *M) {
+    if (F.getCallingConv() == CallingConv::SPIR_KERNEL)
+      Work.push_back(&F);
+  }
+  for (auto &F : Work) {
```

Comment:
Sorry, I've just thought about this little improvement. It's just a nit:
```suggestion
  for (auto &F : *M) {
    // Skip functions that have not a SPIR kernel calling conv
    if (F.getCallingConv() != CallingConv::SPIR_KERNEL)
      continue;
```

Reply:
I'll apply this comment after the original request is merged.

Reply:
Not sure if that's a valid suggestion, as the loop may modify the Module's function list (using `getOrCreateFunction`) while its being iterated over.